quic/s2n-quic-core/src/crypto/tls/null_tls_context.rs

```diff
@@ -0,0 +1,510 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Minimal endpoint that doesn't perform any handshakes or encryption
+//!
+//! NOTE: this should only be used for internal testing
+//!
+//! The main purpose of this endpoint type is to identify bottlenecks in the code _outside_ of
+//! crypto, since that's a large portion of the cycles we use. This has similar goals to the
+//! proposal in https://datatracker.ietf.org/doc/html/draft-banks-quic-disable-encryption-00, but
+//! instead reduces the handshake to simply exchanging transport parameters.
+
+use crate::{
+    application::{server_name::LOCALHOST, ServerName},
+    crypto::{self, tls},
+    transport,
+};
+use bytes::Bytes;
+use core::{mem::size_of, task::Poll};
+
+#[derive(Debug)]
+pub struct Endpoint(pub UserProvidedTlsContext);
```

Comment:
I don't think we need a new endpoint just for this. I think it would be better to have the existing null TLS provider take an `Option<Box<dyn Any>>`

Reply:
We need clone it(`new_server_session` take `&mut self`), But Clone isn't dyn compatible, Box<dyn Any + Clone> don't works

Reply:
I rewrite to use generic over T, you can decide accept generic or keep original